pkg/dockerfile/dockerfile.go

```diff
@@ -32,10 +34,23 @@ import (
 
 // Stages parses a Dockerfile and returns an array of KanikoStage
 func Stages(opts *config.KanikoOptions) ([]config.KanikoStage, error) {
-	d, err := ioutil.ReadFile(opts.DockerfilePath)
+	var err error
+	var d []uint8
+	match, _ := regexp.MatchString("^https?://", opts.DockerfilePath)
+	if match {
+		response, err := http.Get(opts.DockerfilePath)
```

Comment:
I think this is the issue - you have err defined on line 37, then this redefines it in the new scope. You set err again below on 45 and 47, but don't check until 50. The checked err is outside this scope.

One fix would be to check the errors on 45 and 47 right away, or you could define response separately:

```golang
var response *http.Response
response, err = http.Get(opts.Dockerfile)
```